### PR TITLE
[SYCL] Fix windows build and inference

### PR DESCRIPTION
**Pull request by luoyu-intel**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

Changes:
1. Fix SYCL exception when compiled with win-debug, caused by linking release sycl library.
2. Add sycl debug and release presets
3. Fix crash on Windows
4. Remove sycl::context, sycl::queue is enough

A770 on Windows:
```
Building a website can be done in 10 simple steps:
 Einzelnes
Step 1: Define the purpose of your website. What is the main reason you want to build a website? Is it to sell products, provide information, or connect with customers? Knowing the purpose of your website will help you determine the design and content.
Step 2: Choose a domain name. Your domain name is the web address people will use to access your website. Choose a name that is easy to remember, relevant to your business, and available.
Step 3: Select a web hosting service. Your web hosting service will store your website's files and make them accessible to

llama_print_timings:        load time =   27234.27 ms
llama_print_timings:      sample time =       4.80 ms /   128 runs   (    0.04 ms per token, 26638.92 tokens per second)
llama_print_timings: prompt eval time =   24166.25 ms /    14 tokens ( 1726.16 ms per token,     0.58 tokens per second)
llama_print_timings:        eval time =    3548.60 ms /   127 runs   (   27.94 ms per token,    35.79 tokens per second)
llama_print_timings:       total time =   27780.29 ms /   141 tokens
```

### CODE REVIEW

File: ggml.h [hunk 1]
```diff
@@ -312,6 +312,12 @@
     GGML_TENSOR_LOCALS(int64_t, ne,  dst,  ne) \
     GGML_TENSOR_LOCALS(size_t,  nb,  dst,  nb)
 
+#define GGML_TENSOR_BINARY_OP_LOCALS01 \
```
Comment (airMeng) [inline ggml.h:315]: what does ```S01``` mean?

Comment (luoyu-intel) [reply ggml.h:315]: only src0 and src1, no dst

**Review by airMeng — APPROVED** [review]

**Comment by NeoZhangJianyu** [comment]
1. Have the DPCT helper functions been verified  on Linux?

2. What's the impact of performance with this PR?

3. The Ninja is used to build windows.
Could you update the windows chapters in README-sycl.md?
a. add Ninja installation.
b. rm mingw-w64 make installation.
c. update how to build release/debug.

**Comment by luoyu-intel** [comment]
Hi @NeoZhangJianyu 
> 1. Have the DPCT helper functions been verified  on Linux?

Yes, all UTs are passed. The model inference is correct.
> 2. What's the impact of performance with this PR?

No performance change. A770 on windows(this PR) has the same performance as A770 on Linux(master)
> 3. The Ninja is used to build windows.
>    Could you update the windows chapters in README-sycl.md?
>    a. add Ninja installation.
>    b. rm mingw-w64 make installation.
>    c. update how to build release/debug.

Yes, I will update the README. Visual Studio integrates Ninja as default. There are no extra installation steps for it.

**Comment by airMeng** [comment]
@ggerganov does CI often encounter some issues recently?

**Comment by luoyu-intel** [comment]
@NeoZhangJianyu README has been updated.

**Review by NeoZhangJianyu — APPROVED** [review]
It's good to enhance for Windows!

**Review by airMeng — APPROVED** [review]

**Pull request merged**